jetty-websocket/websocket-core-common/src/main/java/org/eclipse/jetty/websocket/core/Extension.java

```diff
@@ -13,20 +13,23 @@
 
 package org.eclipse.jetty.websocket.core;
 
+import java.io.Closeable;
+import java.io.IOException;
+
 /**
  * Interface for WebSocket Extensions.
  * <p>
  * That {@link Frame}s are passed through the Extension via the {@link IncomingFrames} and {@link OutgoingFrames} interfaces
  */
-public interface Extension extends IncomingFrames, OutgoingFrames
+public interface Extension extends IncomingFrames, OutgoingFrames, Closeable
 {
 
     void init(ExtensionConfig config, WebSocketComponents components);
 
     /**
      * Used to clean up any resources after connection close.
      */
-    default void close()
+    default void close() throws IOException
```

Comment:
Drop the `throws IOException` clause, you want those methods to never throw (if they throw, the only choice is to log and ignore anyway).

Reply:
If I drop `IOException` here then none of the `Extension` implementations can declare that it throws `IOException`. And then why do I need to call `IO.close()` since an extension cannot declare that it throws `IOException`, the try/catch in `IO.close` is useless and I could just call `ext.close()` directly.

Reply:
I thought IO.close() would catch Throwable, but I was wrong.
Remove the throws, and keep the try/catches in ExtensionStack.